neuromation/api/storage.py

```diff
@@ -735,67 +785,23 @@ def _file_status_from_api_stat(cluster_name: str, values: Dict[str, Any]) -> Fil
 ProgressQueueItem = Optional[Tuple[Callable[[Any], None], Any]]
 
 
-class QueuedProgress:
```

Comment:
`QueuedProgress` was already little bit wrongly typed: it was used for both `AbstractFileProgress` and `AbstractRecursiveFileProgress` with casts like the following:
```
progress = cast(AbstractRecursiveFileProgress, self._progress)
```
Moreover, `AbstractDeleteProgress` is not subclass of `AbstractFileProgress`, so trying to reuse it would lead to even more strange typing. So I decided to generalize the pattern (put calls into `asyncio.Queue`) to make it easy to reuse it for my case and make for solid typing by adding `_AsyncAbstract...` classes. Initially I thought that I will be able to do proper annotations using generics, but I don't know any way to say mypy that function returns almost same type but all methods are replaced with coroutines.

Reply:
Then I think that it would be better to split this PR on two parts. One implements the rm progress and other refactors the progress machinery.

Reply:
PR for refactoring the progress machinery: https://github.com/neuromation/platform-client-python/pull/1667
Sorry that I didn't do this initially.